src/Kiota.Builder/Writers/Python/CodeClassDeclarationWriter.cs

```diff
@@ -49,20 +47,40 @@ public override void WriteCodeElement(ClassDeclaration codeElement, LanguageWrit
         }
     }
 
-    private void WriteInternalImports(ClassDeclaration codeElement, LanguageWriter writer, CodeNamespace parentNameSpace) {
+    private static void WriteInternalImports(ClassDeclaration codeElement, LanguageWriter writer) {
         var internalImportSymbolsAndPaths = codeElement.Usings
                                                         .Where(x => !x.IsExternal)
-                                                        .Select(x => _relativeImportManager.GetRelativeImportPathForUsing(x, parentNameSpace))
+                                                        .Select(x => GetImportPathForUsing(x))
                                                         .GroupBy(x => x.Item3)
                                                         .Where(x => !string.IsNullOrEmpty(x.Key))
                                                         .OrderBy(x => x.Key);
         if(internalImportSymbolsAndPaths.Any()){
             foreach (var codeUsing in internalImportSymbolsAndPaths)
-                writer.WriteLine($"from {codeUsing.Key.ToSnakeCase()} import {codeUsing.Select(x => GetAliasedSymbol(x.Item1, x.Item2)).Distinct().OrderBy(x => x).Aggregate((x,y) => x + ", " + y)}");
+                foreach(var symbol in codeUsing.Select(x => GetAliasedSymbol(x.Item1, x.Item2)).Distinct().OrderBy(x => x))
+                    writer.WriteLine($"{symbol} = lazy_import('{codeUsing.Key.ToSnakeCase().Replace("._", ".")}.{symbol}')"); 
             writer.WriteLine();
         }
     }
     private static string GetAliasedSymbol(string symbol, string alias) {
         return string.IsNullOrEmpty(alias) ? symbol : $"{symbol} as {alias}";
     }
+
+        /// <summary>
+    /// Returns the import path for the given using and import context namespace.
+    /// </summary>
+    /// <param name="codeUsing">The using to import into the current namespace context</param>
+    /// <returns>The import symbol, it's alias if any and the import path</returns>
+    private static (string, string, string) GetImportPathForUsing(CodeUsing codeUsing)
+    {
+        if (codeUsing?.IsExternal ?? true)
+            return (string.Empty, string.Empty, string.Empty);//it's an external import, add nothing
+        var typeDef = codeUsing.Declaration.TypeDefinition;
+        if (typeDef == null)
+            return (null, codeUsing.Alias, ""); // it's relative to the folder, with no declaration (default failsafe)
+        
+        var importSymbol = codeUsing.Declaration == null ? codeUsing.Name : typeDef.Name.ToSnakeCase();
```

Comment:
if `codeUsing.Declaration` can be null here, that means we can get a NullReferenceException on line 77 where we ty to access `codeUsing.Declaration.TypeDefinition`.

Reply:
Updated in https://github.com/microsoft/kiota/pull/2028/commits/3921062b13880b666d91c9378513b88676d7e223